test/git_issue_898.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright Matt Borland, 2022
+// Use, modification and distribution are subject to the
+// Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <cmath>
+#include <limits>
+#include <boost/math/quadrature/tanh_sinh.hpp>
+#include <boost/math/constants/constants.hpp>
+#include "math_unit_test.hpp"
+
+// numerically evaluate the integral for Stefan-Boltzmann Law from Planck's Law
+// https://en.wikipedia.org/wiki/Stefan%E2%80%93Boltzmann_law#Derivation_from_Planck's_law
+template <typename Real>
+Real StefanBoltzmann()
+{
+    constexpr auto h = Real(6.62607015e-34);
+    constexpr auto c = Real(299792458.0);
+    constexpr auto kB = Real(1.380649e-23);
+
+    constexpr auto c1 = Real(2) * boost::math::constants::pi<Real>() * h * c * c ;
+    constexpr auto c2 = h * c / kB;
+    constexpr auto T = 1000.;
```

Comment:
Interestingly, making T a double here *does not* reproduce the issue, making it an int:

```
constexpr auto T = 1000;
```

Not only reproduces the issue, but leads on to an out of bounds array subscript later on.

BTW, I'm not convinced the suggested fix is quite correct: the purpose of `max_left_position` is to protect against evaluation at a singularity (or in this case, an infinite result).  This might need some careful thought...

Reply:
If `max_left_position` is 0 you would get underflow to `std::numeric_limits<size_t>::max()`, and then the array index is `max_left_position + 1` which overflows back to 0, but avoids seg fault.